src/app/components/Upload/Upload.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+import { uploadHighlight } from '../../utils/highlights';
+
+type UploadProps = {
+  src: string;
+};
+
+function Upload({ src }: UploadProps): JSX.Element {
+  const [url, setUrl] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [total, setTotal] = useState(0);
+  const [loaded, setLoaded] = useState(0);
+
+  async function handleUpload() {
+    try {
+      const url = await uploadHighlight({
+        src,
+        onProgress: ({ loaded, total }) => {
+          setTotal(total);
+          setLoaded(loaded);
+        },
+      });
+      setUrl(url);
+    } catch (error) {
+      console.error(error);
+      setErrorMessage(error.message);
+    }
```

Comment:
```suggestion
    } catch (error) {
      console.error(error);
     if (error instanceof Error) {
        setErrorMessage(error.message);
      }
    }
```

just to be sure, as discussed yesterday